x/btcstaking/keeper/btc_reorg.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+)
+
+// HaltIfBtcReorgLargerThanConfirmationDepth safety mechanism to stop the chain in case there is an BTC reorg
+// higher than the BtcConfirmationDepth. In teory this should only happen if the babylon chain goes down for
+// a period longer than (2 * BtcConfirmationDepth * 10min).
+func (k *Keeper) HaltIfBtcReorgLargerThanConfirmationDepth(ctx context.Context) {
+	p := k.btccKeeper.GetParams(ctx)
+
+	largestReorg := k.MustGetLargestBtcReorg(ctx)
+	if largestReorg > p.BtcConfirmationDepth {
```

Comment:
just to double check.

We are allowing stakes which are included in blocks at confirmation depth: https://github.com/babylonlabs-io/babylon/blob/v1.0.0-rc.5/x/btcstaking/keeper/inclusion_proof.go#L60, so maybe we should panic here if
```
largestReorg >= p.BtcConfirmationDepth
```
i.e main reason for this fail safe is that we won't progress forward with stake that potentially has been reorged.

Maybe we can add some more comprehensive test in `msg_server_test.go` which add some stake on depth boundary and trigger this fail safe to see if such stake would be affected ?